### DISCUSSION TITLE
Issue with Laravel $casts when JSON is empty, resulting in null instead of an empty array

### DISCUSSION THREAD

**Discussion by romdh4ne** [laravel/framework#49086] [Ideas]
### Laravel Version

10.10

### PHP Version

8.2.12

### Database Driver & Version

MySQL

### Description

Dear Laravel Development Team,

I hope this message finds you well. I am writing to bring attention to an issue I have encountered with the $casts functionality in Laravel, specifically when dealing with empty JSON fields. Currently, when a JSON field is empty, the $casts attribute seems to be returning null instead of an empty array, which is the expected behavior when casting to an array.

**Issue Description:**
When utilizing $casts in a Laravel model to **cast a JSON field to an array**, the expected behavior is that an empty JSON field should be cast to an empty array. However, in the current implementation, **an empty JSON field results in the casted attribute being set to null instead of an empty array**.

**Example:**
Consider the following model definition:

```
class ExampleModel extends Model
{
    protected $casts = [
        'json_field' => 'array',
    ];
}
```
**If the 'json_field' is empty in the database, accessing it on an instance of ExampleModel would ideally return an empty array. However, it currently returns null.**

### Steps To Reproduce

1. Create a model with a JSON field.

1. Set $casts for the JSON field to 'array'.

1. Save an instance of the model with an empty JSON field.

1. Retrieve the model instance and access the casted attribute.

**Expected Result:**
The casted attribute should be an empty array.

**Actual Result:**
The casted attribute is currently set to null.

**Comment by ghost** [comment]
Hi.
From documentation 

Attributes that are null will not be cast.
https://laravel.com/docs/10.x/eloquent-mutators#array-and-json-casting

Also, json_decode('', true) returns null.

**Reply by ghost** [reply]
That's true , thank u !
it can be an idea for the next version of laravel :)
